Docs: Update Loki documentation

ivanahuckova opened the pull request:
**What this PR does / why we need it**:
Updates documentation for Loki data source

ivanahuckova committed:
Update Loki documentation

ivanahuckova committed:
Update

osg-grafana commented:
@achatterjee-grafana, can you add the Docs Squad as a code owner?

achatterjee-grafana commented:
@osg-grafana done. I am working with Ivana on this PR, and also David has a section to his name, that is why we got added. (Clarification).

oddlittlebird commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+Querying and displaying log data from Loki is available via [Explore]({{< relref "../explore" >}}), and with the [logs panel]({{< relref "../panels/visualizations/logs-panel.md" >}}) in dashboards. Select the Loki data source, and then enter a [LogQL](https://grafana.com/docs/loki/latest/logql/#log-queries) query to display your logs.
 
-`{job="mysql"} |= "error" != "timeout"`
+A log query consists of two parts: **log stream selector**, and a **log pipeline**. For performance reasons you need to start by choosing a log stream by selecting a log label.
```
Why are the terms bold? If they are not on the UI, they should not be bold. Apply italic if you want to emphasize them.

ivanahuckova replied:
Removed bold. I guess, they don't need to be bold.

oddlittlebird commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+A log query consists of two parts: **log stream selector**, and a **log pipeline**. For performance reasons you need to start by choosing a log stream by selecting a log label.
 
-The following filter types are currently supported:
+### Log Context
```
```suggestion
### Log context
```

oddlittlebird commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
-- `!=` line doesn't contain string.
-- `|~` line matches regular expression.
-- `!~` line does not match regular expression.
+When using a search expression as detailed above, you now have the ability to retrieve the context surrounding your filtered results.
```
```suggestion
When using a search expression as detailed above, you can retrieve the context surrounding your filtered results.
```

oddlittlebird reviewed (changes requested):
Great changes! Here are some of my suggestions.

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
+This topic explains options, variables, querying, and other options specific to this data source. Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.
 
-## Adding the data source
+> **Note:** If you have issues with getting this data source to work and Grafana is giving you undescriptive errors, then check your log file (try looking in /var/log/grafana/grafana.log).
```
```suggestion
> **Note:** To troubleshoot configuration and other issues, check the log file (/var/log/grafana/grafana.log).
```

oddlittlebird replied:
This path is not going to work for all users. On my Windows machine, it's in grafana/data/log/grafana.log. I don't have a var folder.

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
+| `Default`             | Default data source means that it will be pre-selected for new panels.                                                                        |
+| `URL`                 | The URL of the Loki instance, e.g., `http://localhost:3100`                                                                                   |
+| `Whitelisted Cookies` | Grafana Proxy deletes forwarded cookies by default. Specify cookies by name that should be forwarded to the data source.                      |
+| `Maximum lines`       | Upper limit for number of log lines returned by Loki (default is 1000). Decrease if your browser is sluggish when displaying logs in Explore. |
```
```suggestion
| `Maximum lines`       | Upper limit for the number of log lines returned by Loki (default is 1000). Lower this limit if your browser is sluggish when displaying logs in Explore. |
```

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+There are two types of LogQL queries:
 
-The [same rules that apply for Prometheus Label Selectors](https://prometheus.io/docs/prometheus/latest/querying/basics/#instant-vector-selectors) apply for Loki Log Stream Selectors.
+- **Log queries** that return the contents of log lines.
```
```suggestion
- **Log queries:** returns the contents of log lines.
```

achatterjee-grafana replied:
Log queries do not need to be bold.

ivanahuckova replied:
Removed bold. Plus this was changed to `Log queries -`

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
 
-The [same rules that apply for Prometheus Label Selectors](https://prometheus.io/docs/prometheus/latest/querying/basics/#instant-vector-selectors) apply for Loki Log Stream Selectors.
+- **Log queries** that return the contents of log lines.
+- **Metric queries** that extend log queries and calculate sample values based on the content of logs from a log query.
```
```suggestion
- **Metric queries:** extends log queries and calculate sample values based on the content of logs from a log query.
```

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+Querying and displaying log data from Loki is available via [Explore]({{< relref "../explore" >}}), and with the [logs panel]({{< relref "../panels/visualizations/logs-panel.md" >}}) in dashboards. Select the Loki data source, and then enter a [LogQL](https://grafana.com/docs/loki/latest/logql/#log-queries) query to display your logs.
 
-`{job="mysql"} |= "error" != "timeout"`
+A log query consists of two parts: **log stream selector**, and a **log pipeline**. For performance reasons you need to start by choosing a log stream by selecting a log label.
```
```suggestion
A log query consists of two parts: **log stream selector**, and a **log pipeline**. For performance reasons begin by choosing a log stream by selecting a log label.
```

achatterjee-grafana reviewed (changes requested):
Added some comments. Will add the rest ASAP.

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+log message you're interested in.
 
-> **Note:** For more details about LogQL, Loki's query language, refer to the [Loki LogQL](https://grafana.com/docs/loki/latest/logql/)
+> **Note:** This feature is only available in Grafana v6.3+
```
Can we remove this note? 6.3 is really old.

ivanahuckova replied:
removed

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
-> Grafana v6.x comes with Explore enabled by default.
-> In Grafana v5.3.x and v5.4.x. you need to enable Explore manually.
-> Viewing Loki data in dashboard panels is supported in Grafana v6.4+.
+Grafana ships with built-in support for Loki, an open source log aggregation system by Grafana Labs. Add it as a data source and you are ready to build dashboards or query your log data in [Explore]({{< relref "../explore" >}}).
```
```suggestion
Grafana ships with built-in support for Loki, an open source log aggregation system by Grafana Labs. This topic explains options, variables, querying, and other options specific to this data source.

Add it as a data source and you are ready to build dashboards or query your log data in [Explore]({{< relref "../explore" >}}). Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.
```

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
 
-Grafana ships with built-in support for Loki, Grafana's log aggregation system.
-Just add it as a data source and you are ready to query your log data in [Explore]({{< relref "../explore" >}}).
+This topic explains options, variables, querying, and other options specific to this data source. Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.
```
Move content up, as shown.

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+## Loki Query Editor
 
-Querying and displaying log data from Loki is available via [Explore]({{< relref "../explore" >}}), and with the [logs panel]({{< relref "../panels/visualizations/logs-panel.md" >}}) in dashboards. Select the Loki data source, and then enter a [LogQL](https://grafana.com/docs/loki/latest/logql/) query to display your logs.
+Loki query editor can be used to create both log and metric queries.
```
```suggestion
Use the Loki query editor to create both log and metric queries.
```

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
-`{app="mysql",name="mysql-backup"}`
+| Name               | Description                                                                                                                                                                                                 |
+| ------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Query expression` | Loki query expression, check out the [LogQL documentation](https://grafana.com/docs/loki/latest/logql/).                                                                                                    |
```
```suggestion
| `Query expression` | Loki query expression, refer to the [LogQL documentation](https://grafana.com/docs/loki/latest/logql/) for more information.                                                                                                    |
```

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
+| Name               | Description                                                                                                                                                                                                 |
+| ------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Query expression` | Loki query expression, check out the [LogQL documentation](https://grafana.com/docs/loki/latest/logql/).                                                                                                    |
+| `Query type`       | Choose the type of query you would like to run. An instant query queries against a single point in time. We are using "To" time from the time range. A range query queries over the selected range of time. |
```
```suggestion
| `Query type`       | Choose the type of query to run. The instant type queries against a single point in time. We are using "To" time from the time range. The range type queries over the selected range of time. |
```

achatterjee-grafana commented on docs/sources/datasources/loki.md:
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
 
-The [same rules that apply for Prometheus Label Selectors](https://prometheus.io/docs/prometheus/latest/querying/basics/#instant-vector-selectors) apply for Loki Log Stream Selectors.
+- **Log queries** that return the contents of log lines.
+- **Metric queries** that extend log queries and calculate sample values based on the content of logs from a log query.
```
Metric queries do not need to be bold.

achatterjee-grafana reviewed (changes requested):
Some more review comments. Thanks for updating the topic Ivana!

osg-grafana commented:
> @osg-grafana done. I am working with Ivana on this PR, and also David has a section to his name, that is why we got added. (Clarification).

Thank you. I will keep an eye on this content to align on it when working on GEL.

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ivanahuckova committed:
Update docs/sources/datasources/loki.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

ivanahuckova committed:
Merge remote-tracking branch 'origin' into ivana/loki-documentation

ivanahuckova committed:
Update, add images and log browser info

ivanahuckova committed:
Add documentation for label browser

ivanahuckova commented:
1. I have addressed all feedback.
2. Added log browser documentation: https://github.com/grafana/grafana/pull/31867/commits/1e3bb4d6924a58357602e412c1a608ebc5736ca8
3. New images added in: https://github.com/grafana/website/pull/3814

ivanahuckova commented on docs/sources/datasources/loki.md:
```diff
@@ -49,97 +43,73 @@ Each derived field consists of:
+With Loki log browser you can easily navigate trough your list of labels and values and construct the query of your choice. Log browser has multi-step selection:
+1. Choose the labels you would like to consider for your search.
+2. Pick the values for selected labels. Log browser supports facetting and therefore it shows you only possible label combinations. 
+3. Choose the type of query - logs query or rate metrics query. Additionaly, you can also validate selector.
```
```suggestion
3. Choose the type of query - logs query or rate metrics query. Additionally, you can also validate selector.
```

ivanahuckova committed:
Update docs/sources/datasources/loki.md

oddlittlebird reviewed (approved):
Great work, Ivana!

achatterjee-grafana reviewed (approved):
Great work!
